### PR TITLE
(Refactor) Use final form step three

### PR DESCRIPTION
**Pull request by fernandomg**
This is another step towards the completion of #711 

In this PR `StepTree` was moved entirely into `final-form`, with the exception of whitelist which is now part of `stepThreeForm` (similar to what was done with `reservedTokens` in `stepTwo`).

I didn't manage to prevent error messages for `rate` and `supply` for the Tiers added after the first one. This won't affect the UX, but will display the errors by default.

Besides I've created an `Error` component, which makes easier to display errors related to `final-form` Fields.

This looks still as a WIP, but the idea is to keep moving toward the adoption of `final-form` as fast as possible.

### CODE REVIEW

**Comment by fernandomg** [comment]
@dennis00010011b  this PR sligtly changed the DOM structure of the Step Three, and is giving errors with e2e tests.

**Comment by dennis00010011b** [comment]
@fernandomg  I've updated e2e script, test for upload CSV should work now.
Try to update submodule token-wizard-test-automation and commit again

File: src/components/stepThree/GasPriceInput.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {this.state.isCustom ? (
+          <input
+            type="number"
+            className="input"
+            value={this.state.customGasPrice}
+            onChange={(e) => this.handleCustomGasPriceChange(e.target.value)}
+          />
+        ) : null}
+        <p className="description">Slow is cheap, fast is expensive</p>
+        <Error name="gasPrice"/>
```
Comment (fvictorio) [inline src/components/stepThree/GasPriceInput.js]: The name shouldn't be hardcoded here, right?

File: src/components/Common/Error.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const defaultErrorStyles = {
+  color: 'red',
+  fontWeight: 'bold',
+  fontSize: '12px',
+  width: '100%',
+  height: '20px',
+}
+
+export const Error = ({ name, errorStyle }) => (
+  <Field
```
Comment (fvictorio) [inline src/components/Common/Error.js]: I'm not sure I understand this. Is the `Field` component really necessary? Couldn't we just receive the whole `meta` object from props and use that?

If I'm understanding this correctly, this means that we'll have two `Field`s with the same `name` attribute, but one of them is used only to show errors. It doesn't sound right to me, is there a good reason for this approach?

File: src/components/Common/TierBlock.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                label={SUPPLY}
+                description={DESCRIPTION.SUPPLY}
+              />
+              {
+                /*
+                  * TODO: REVIEW. I'm not sure about this approach.
+                  * But it worked for me to keep the error messages properly updated for the minCap field.
+                  */
+              }
+              <Field name="minCap" subscription={{}}>
```
Comment (fvictorio) [inline src/components/Common/TierBlock.js]: Same here thing (as in the Error component). Is the Field necessary? Can't we use the [`change`](https://github.com/final-form/react-final-form#change-name-string-value-any--void) function?

File: src/components/stepThree/GasPriceInput.js (hunk 1 above)
Comment (fernandomg) [reply src/components/stepThree/GasPriceInput.js]: fixed

**Comment by fernandomg** [comment]
@dennis00010011b I've been running the e2e locally and it is still failing.

This are the lists of issues that I found:
1. For the error messages you use the xpath selector and look for the only attr you had available that is `style`, I'd slightly changed the styles for the `stepThree` and now it's failing. As a result I've just added the `class="error"` to the errors `<p>` tags.
1.  `Wizard step#3: User is able to bulk delete all whitelisted addresses` is failing for me, despite I reproduced the steps in the browser and worked properly.
1.  All test that contains the assert `assert.equal(flagCrowdsale, true);` fail in the `testSuite1.js` (you can see them here: https://travis-ci.org/poanetwork/token-wizard/builds/362808731#L2724)

---

As a note, with this migration to `final-form` we're adding `name` attributes to most of the inputs in the wizard, so it will be easier and cleaner for you to look for a particular tag despite its location in the DOM.

If you find yourself using XPath to reach a particular element, don't hesitate in request a change in the code to add an `id`, a `name` or a `class`, or whatever attribute is needed, to make it easier for you to build the tests.

**Comment by dennis00010011b** [comment]
@fernandomg  Thank you for feedback.
1. I’ll change finding the errors by class name. It would really helpful if elements would have id or at least name or class.
2. I see this issue with bulk delete, still can’t figure out it 
3. Almost all test use pre-created crowdsale . flagCrowdsale used for skip  tests if crowdsale hasn’t created.

**Pull request merged**